frontend/src/api/invest.ts

```diff
@@ -30,23 +30,43 @@ export async function getLoansToInvest(cb: (loans: LoanToInvestT[]) => void): Pr
   cb(loansToInvest);
 }
 
-export async function getLoanToInvest(shortId) {
+export async function getLoanToInvest(shortId: string, cb: (loan: LoanToInvestT) => void) {
   const api = await API.instance();
   let currentUser = await api.currentUser();
   let blockchainLoan = await api.loan(shortId);
   await blockchainLoan.updateStateFromBlockchain();
+  return cb(await blockchainLoanToVueLoan(blockchainLoan));
+}
+
+export async function investInLoan(shortId: string, amount: number) {
+  const api = await API.instance();
+  const testToken = api.testToken;
+  let currentUser = await api.currentUser();
+  let loan = await api.loan(shortId);
+  await loan.invest(await testToken.integerize(new BigNumber(amount)));
+}
+
+export async function getMyInvestments(cb: (loans: LoanToInvestT[]) => void) {
+  const api = await API.instance();
+  let user = await api.currentUser();
+  let blockchainLoans = await api.loansByOwner(user);
+  await Promise.all(blockchainLoans.map(loan => loan.updateStateFromBlockchain()));
+  cb(await Promise.all(blockchainLoans.map(blockchainLoanToVueLoan)));
+}
 
+async function blockchainLoanToVueLoan(blockchainLoan): Promise<LoanToInvestT> {
```

Comment:
Add a type declaration to the argument.

Also, it's not really a blockchain loan, it's something more like Library Loan? And it's not a VueLoan, It's a ViewLoan. I think.

Reply:
right!